### PR TITLE
[Refactor] unravel_key fixes

### PR DESCRIPTION
**Pull request by vmoens**
cc @matteobettini

### CODE REVIEW

**Comment by vmoens** [comment]
interestingly it does not work yet but the benchmarks are better!

File: tensordict/persistent.py [hunk 1]
```diff
@@ -228,10 +228,7 @@ def close(self):
 
     def _process_key(self, key):
         key = unravel_keys(key)
```
Comment (matteobettini) [inline tensordict/persistent.py:230]: one more general thought: shouldn't `unravel_keys` be called `unravel_key`?
cause it always acts on a single key.
especially cause now we have `unravel_key_list` for multiple keys

**Review by matteobettini — APPROVED** [review]
LGTM

File: tensordict/nn/common.py [hunk 2]
```diff
@@ -1087,6 +1090,10 @@ def forward(
                 else:
                     raise err
             if isinstance(tensors, (dict, TensorDictBase)):
+                if isinstance(tensors, dict):
+                    keys = unravel_key_list(list(tensors.keys()))
```
[the remark this answers is not in the document]
Comment (vmoens) [reply tensordict/nn/common.py:1094]: we use multiple dispatch to make it work with tuples and lists (cf c++ function)

But it's also a bit dangerous, as one may quickly end up mixing a tuple of tuples as being a single key or a tuple of tuple as being a list of keys...

File: tensordict/persistent.py (hunk 1 above)
Comment (vmoens) [reply tensordict/persistent.py:230]: yep I agree :) let me do that

**Comment by vmoens** [comment]
Some benchmarks are faster
Almost all stacked benchmarks are slower but that's the price to pay for the bugfix
I'm more worried about things like `test_getitem[slice_int]` which should not be impacted. I suspect that the `try / except` we add in `__getitem__` is slow when the exception is reached. The alternative is to check the content of the index and then call the unravel function but that means going through the index twice...

**Pull request merged**